agent/api/task/task.go

```diff
@@ -675,7 +685,28 @@ func (task *Task) initializeSSMSecretResource(credentialsManager credentials.Man
 				resourcestatus.ResourceStatus(ssmsecret.SSMSecretCreated),
 				apicontainerstatus.ContainerCreated)
 		}
+
+		// Firelens container needs to depends on secret if other containers use secret log options.
+		if container.FirelensConfig != nil && task.firelensDependsOnSecretResource(apicontainer.SecretProviderSSM) {
```

Comment:
`if container.FirelensConfig != nil` would need a lock. Instead why don't we add a method to Container struct called `isFirelensContainer` and do this check there?

Reply:
i ended up adding GetFirelensConfig to protect it with lock since there's some place i want to look at the config